managed/src/main/java/com/yugabyte/yw/models/helpers/CustomerConfigValidator.java

```diff
@@ -179,4 +194,48 @@ public ObjectNode validateDataContent(JsonNode formData) {
         formData.get("data"), errorJson));
     return errorJson;
   }
+
+  /**
+   * Validates data which is contained in formData.
+   * During the procedure it checks all the S3 config parameters whether they are valid or not.
+   * Errors are collected and
+   * returned back as a result. Empty result object means no errors.
+   *
+   * Currently is checked:
+   *  - S3/AWS - S3 Bucket, whether it exists or not.
+   *
+   * @param formData, region
+   * @return Json filled with errors
+   */
+  public static ObjectNode validateS3DataContent(JsonNode formData, ObjectNode errorJson) {
+    try {
+      String bucketname = formData.get("BACKUP_LOCATION").asText();
+      //Assuming bucket name will always start with s3:// otherwise that will be invalid
+      bucketname = bucketname.substring(5);
+      AmazonS3 s3client = getS3Client(formData);
+      List<Bucket> buckets = s3client.listBuckets();
+      for (Bucket bucket : buckets) {
+        if (bucket.getName().equals(bucketname))
+          return errorJson;
+      }
+      errorJson.set("BackupConfigException", Json.newArray().add("Bucket name " + bucketname +
+          " doesn't exist"));
+    } catch (AmazonS3Exception s3Exception) {
+      errorJson.set("BackupConfigException", Json.newArray().add(s3Exception.getErrorMessage()));
```

Comment:
Here is another question - in other places we are using a pattern errorJson.set("fieldname", "error string"). Using the field name UI can select the proper field and mark it as erroneous (later). Your usage is different and doesn't look meaningful.

Reply:
@SergeyPotachev  @sb-yb  How are we going to handle n/w failure, timed out or something erroneous which is not specific to any field. In that case do we want to throw error message in the form of string instead of errorJson Array?

Reply:
It is still a network error during validation of that field. So you should still put it under fieldname.